runtime/lua/vim/lsp/util.lua

```diff
@@ -296,6 +296,9 @@ local function get_lines(bufnr, rows)
   end
 
   local filename = api.nvim_buf_get_name(bufnr)
+  if vim.fn.isdirectory(filename) ~= 0 then
```

Comment:
For my own learning: It feels odd for a function to be named `isX` without returning a boolean. Is this naming something that we have control over or is part of our vim legacy?

Reply:
It's because in vimscript boolean `1` is true and `0` is false, but in lua this doesn't work as both of these are truthy. I've suggested adding a layer for vimscript boolean functions to return true and false instead of 1 and 0 but it was shot down due to adding extra magic between the vimscript<->lua layer (which I can't disagree with completely). It's not an optimal situation, I agree. I wonder if we couldn't supplement `vim.fs` with a function like this, since I feel like "is directory" is a useful and common enough check to get it's own dedicated function. Something like `vim.fs.isdir`?

Reply:
> Something like `vim.fs.isdir`?

Yeah we need that, though maybe `vim.fs.ftype() => 'file'|'dir'|...` would be better.

Reply:
I would like to make a case against `vim.fs.ftype()`. I understand the API surface would be reduced, but I have a suspicion separate `isdir` and `isfile` functions would be more useful. My reasoning is that we already have something like `vim.fs.ftype` which is `vim.uv.fs_stat`. Despite this, both [neovim](https://github.com/neovim/neovim/blob/master/test/testutil.lua#L34-L43) and [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/util.lua#L117-L132) have created isdir/isfile helper functions. I think this points to there being a value in having separate isdir/isfile functions.